lib/ood_core/job/adapters/lsf.rb

```diff
@@ -89,7 +89,14 @@ def submit(script, after: [], afterok: [], afternotok: [], afterany: [])
         def info(id)
           # TODO: handle job arrays
           job = batch.get_job(id: id)
-          job ? info_for_batch_hash(job) : nil
+          if job
+            info_for_batch_hash(job)
+          else
+            Info.new(
+              id: id,
+              status: :completed
+            )
```

Comment:
Is it guarenteed that if job is nil the job is always completed and that other error cases we properly catch and raise? I'm assuming this is similar to how Torque works and is catching the case where the job is no longer in the queue.

Note it appears the other implementations of `#info` could possibly return nil as they all use detect which can return nil if not found. Though I didn't look further into those implementations to see.

Reply:
The job will only be set `nil` if `bjobs` returns an empty string and a successful exit code. Unfortunately, I am not sure how to exhaustively test this.

Some example cases:

1. Job that completed:

   ```console
   $ bjobs -a -w -W 1 2>/dev/null
   $ echo $?
   0
   ```

2. Job that was never created:

   ```console
   $ bjobs -a -w -W 1000000 2>/dev/null
   $ echo $?
   0
   ```

3. A job id that is impossible to exist:

   ```console
   $ bjobs -a -w -W 1000000000000000 2>/dev/null
   $ echo $?
   255
   ```

   ```console
   $ bjobs -a -w -W -1 2>/dev/null
   $ echo $?
   255
   ```

   ```console
   $ bjobs -a -w -W 10sdoijhf9432 2>/dev/null
   $ echo $?
   255
   ```

The first two test cases would return a completed job, whereas the "impossible" job id test cases would raise an error. I think that is reasonable.

I need to look at the other `#info` methods.

Reply:
All the other `#info` methods look good.